chain/src/transaction.rs

```diff
@@ -81,11 +96,15 @@ impl HeapSizeOf for TransactionOutput {
 
 #[derive(Debug, PartialEq, Default, Clone)]
 pub struct Transaction {
+	pub overwintered: bool,
```

Comment:
won't enum be more suitable for `Transaction`?

Reply:
I've tried `enum` earlier with `BTC` + `BCH` pair, it turned out that it is much easier (requires less code) to deal with `struct` + `Option` there, instead of `enum`. You'll end up with struct that has `fn`-s returning `Option`-s anyway.